### PR TITLE
Cross-compilation support

### PR DESCRIPTION
**Pull request by bjoto**
Cross-compilation support with riscv64 as the first target.

This PR has a dependency on https://github.com/kernel-patches/vmtest/pull/194 which also has more information.

### CODE REVIEW

File: rootfs/Makefile [hunk 1]
```diff
@@ -8,4 +8,5 @@ all:
 			sudo ./mkrootfs_debian.sh --arch "$${arch}" --distro "$${distro}" ; \
 		done \
 	done
+	sudo ./mkrootfs_ubuntu.sh --arch riscv64 --distro jammy
```
Comment (chantra) [inline rootfs/Makefile:11]: why not just aligning on the distro? As ,uch as CI will be concerned, having different rootfs distro will be a nightmare. There is #84 which is a separate item to handle more rootfs distros.

File: rootfs/mkrootfs_ubuntu.sh [hunk 2]
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
```
Comment (chantra) [inline rootfs/mkrootfs_ubuntu.sh:1]: I believe `mkrootfs_debian.sh` can handle creating jammy distro just fine. I don't think this script is needed.

File: build-samples/build_samples.sh [hunk 3]
```diff
@@ -15,6 +15,23 @@ if [ $? -eq 0 ]; then
 	export LLVM="-$LLVM_VER"
 fi
 
+if [[ $(uname -m) != "$TARGETARCH" ]]; then
```
Comment (chantra) [inline build-samples/build_samples.sh:18]: I don't think TARGETARCH has been assigned yet. Probably missing TARGETARCH=$4 earlier in the script.

File: build-samples/build_samples.sh [hunk 4]
```diff
@@ -15,6 +15,23 @@ if [ $? -eq 0 ]; then
...
+	aarch64)
+		linuxarch="arm64"
+		;;
+	*)
+		;;
+	esac
+	export ARCH="$linuxarch"
+	export CROSS_COMPILE="$TARGETARCH-linux-gnu-"
+fi
+
```
Comment (chantra) [inline build-samples/build_samples.sh:34]: This is probably worth putting in helpers.sh so it can be re-used across the different actions uniformally.

**Review by chantra — COMMENTED** [review]
Hi @bjoto , sorry for the late update on this one.

This will likely need to be tested with vmtest against your libbpf/ci repo to make sure it works, will comment there separately.

I know this is still a draft, but regardless here are some feedback.

**Comment by bjoto** [comment]
> Hi @bjoto , sorry for the late update on this one.

No worries; As you noted, it's a draft, but I'm still very happy to get comments; Especially general ones!

> This will likely need to be tested with vmtest against your libbpf/ci repo to make sure it works, will comment there separately.
> 
> I know this is still a draft, but regardless here are some feedback.

ACK, and thank you.

File: rootfs/Makefile (hunk 1 above)
Comment (bjoto) [reply rootfs/Makefile:11]: You mean align the other builds to use Ubuntu (Jammy or later)? RISC-V is still missing stable Debian support, so currently Ubuntu Jammy and later is the only option for RV. 

I agree that having different distros is not desirable.

Comment (chantra) [reply rootfs/Makefile:11]: > You mean align the other builds to use Ubuntu (Jammy or later)? RISC-V is still missing stable Debian support, so currently Ubuntu Jammy and later (or Debian Sid) is the only option for RV.
> 
> I agree that having different distros is not desirable.

That would work once we move the runners to Jammy, which is not the case.
At the moment, we build for Debian Bullseye on Ubuntu Focal.... it happens to work, but this is by luck. Jammy target would not work. The problem is due to build dependencies changes between host and target OS. #83 explains the problem.

**Comment by chantra** [comment]
@bjoto I have renamed `master` branch to `main` and will delete `master`. By doing so, this PR will be closed and it does not seem there is a way to change the PR target branch. Please re-open against main at your convenience.

**Comment by chantra** [comment]
relaying this here: https://github.com/kernel-patches/vmtest/pull/194#issuecomment-1947033578

**Pull request closed without merging**